activerecord/test/cases/tasks/postgresql_rake_test.rb

```diff
@@ -228,14 +237,16 @@ def setup
 
     def test_structure_load
       filename = "awesome-file.sql"
-      Kernel.expects(:system).with("psql -q -f #{filename} my-app-db")
+      args = [['psql', 'psql',], '-q', '-f', filename, 'my-app-db']
+      Kernel.expects(:system).with(*args).returns(true)
 
       ActiveRecord::Tasks::DatabaseTasks.structure_load(@configuration, filename)
     end
 
     def test_structure_load_accepts_path_with_spaces
       filename = "awesome file.sql"
-      Kernel.expects(:system).with("psql -q -f awesome\\ file.sql my-app-db")
+      args = [['psql', 'psql',], '-q', '-f', 'awesome\\ file.sql', 'my-app-db']
+      Kernel.expects(:system).with(*args).returns(true)
```

Comment:
It's not practical for these tests to actually run real commands, I don't think.

And yes, the problem is specific to your change: the space in the filename should not be escaped.